src/socket.rs

```diff
@@ -1029,6 +1024,12 @@ impl Inner {
             return Ok(());
         }
 
+        if !self.listener_open || self.connections.len() >= MAX_CONNECTIONS_PER_SOCKET {
```

Comment:
Noted. The scenario it helps in is when you have a `UtpListener` being used to accept a single connection before being thrown away. Once it receives a SYN the listener closes and so any further SYNs for the same connection get thrown away. This is bad if the SYN-ACK gets lost because the connecting peer will keep sending their SYN waiting for a response but the receiving peer won't see the SYNs and so won't reply with SYN-ACKs. So the check for if we already have a connection with that id needs to happen before the check for whether we're accepting new connections.